openapi_python_client/parser/properties/__init__.py

```diff
@@ -423,13 +427,22 @@ def build_union_property(
     *, data: oai.Schema, name: str, required: bool, schemas: Schemas, parent_name: str
 ) -> Tuple[Union[UnionProperty, PropertyError], Schemas]:
     sub_properties: List[Property] = []
+    inverted_mappings = {}
+    for k, v in (data.discriminator.mapping if data.discriminator else {}).items():
+        inverted_mappings[Reference.from_ref(v).class_name] = k
```

Comment:
AD: Check for existence, error if so, but we might want to support that case in the future (several mapped names can reference the same type, e.g. for deprecation purposes)